plugins/zenoh-plugin-storage-manager/src/replica/storage.rs

```diff
@@ -407,10 +406,8 @@ impl StorageService {
                     // track of the Latest value, the timestamp is indeed more recent (it was
                     // checked before being processed): we update our internal structure.
                     if self.capability.history == History::Latest {
-                        self.latest_updates
-                            .lock()
-                            .await
-                            .insert(stripped_key, sample_to_store_timestamp);
+                        latest_updates_guard.insert(stripped_key, sample_to_store_timestamp);
+                        drop(latest_updates_guard);
```

Comment:
I think the `drop()` should be closed after the `if` block, rather than inside.
Otherwise, if history is not `Latest`, it's kept for nothing during the log propagation.

Reply:
It was actually a (wrong) conscious choice because in the on-going refactor the Replication can only be enabled with `History::Latest`… But you need to have that context to know there could only be a propagation if it enters the `if` where the `drop` is. Fixed 💯